### ISSUE TITLE
ViTLP-base

### ISSUE THREAD

**Issue by CourageKeenTriumph** [Veason-silverbullet/ViTLP#4]
I was guided by the donut, and I found you did a fantastic job, which was part of my assumption. I hope to make some innovations on the basis of your work, but due to the scarcity of computing resources in the university, I prefer to get your ViTLP-base model. Is that possible?
Thank you again for your excellent work, which gave me something to think about！

**Comment by Veason-silverbullet** [comment]
Hi, @CourageKeenTriumph I am sorry I cannot provide ViTLP-base because I've already left the company. Due to the company's policy, we had decided to release ViTLP-medium only.

Nevertheless, I encourage using ViTLP-medium as we believe scaling up generative models is an inevitable trend for the academic community and industry. In terms of size, ViTLP-base is 253M vs. ViTLP-medium 380M, the scale-up is incremental.

Also, using Deepspeed Zero-2/3 can help reduce memory usage for training, for which 2~4 Nvidia-3090 (24G) can support fine-tuning (I have validated it at the university's server), though it needs gradient accumulation as a trade-off. Please refer to [https://github.com/Veason-silverbullet/ViTLP/tree/main/finetuning](https://github.com/Veason-silverbullet/ViTLP/tree/main/finetuning) for fine-tuning ViTLP, and thanks for your interest in it.

**Comment by CourageKeenTriumph** [comment]
Thank you for your reply and suggestions. it doesn't matter. Your work has been excellent.
May I know your Pre-training resources? donut's training resources are large, but you use a Multi-segment pre-training Scheme, which makes it possible to train on 24G graphics cards? Maybe it takes a little more training time? Of course gradient accumulation is also necessary, but I wonder if similar pre-training can be done on 4090?

**Comment by Veason-silverbullet** [comment]
The GPU requirement for pre-training and fine-tuning differs by a very large margin.

- For pre-training, I used 32 V100 (32G) with Deepspeed Zero-2 on 4 distributed nodes. The training time was around 3 weeks. I suggest NOT using 4090 for pre-training because it does not support Nvlink. Gradient synchronization is a bottleneck for pre-training, for which we usually use V100/A100/H100 series with Nvlink to handle instead of consumer GPUs like 4090.

- For fine-tuning, 4 Nivdia-3090 is enough. However, it is quite slow because large gradient accumulation steps are needed. I suggest 8 V100 for fine-tuning DocVQA if applicable.

Another noteworthy point is that I had to train ViTLP in FP16 but not BF16. This is because, at that time, V100s were the only GPUs I could use, which didn't support BF16. Using BF16 for pre-training is a default choice for modern pre-training, as it is more stable in numeric. Using FP16 caused many overflows during pre-training and impaired model performance. This made ViTLP imperfect (and caused regret to me).
